lib/project_types/script/layers/infrastructure/languages/assemblyscript_task_runner.rb

```diff
@@ -31,14 +37,32 @@ def metadata_file_location
           end
 
           def library_version(library_name)
-            output = JSON.parse(CommandRunner.new(ctx: ctx).call("npm -s list --json"))
+            output = JSON.parse(npm_run("npm -s list --json"))
             library_version_from_npm_list(output, library_name)
           rescue Errors::SystemCallFailureError => error
             library_version_from_npm_list_error_output(error, library_name)
           end
 
+          def set_npm_config
+            npm_run(NPM_SET_REGISTRY_COMMAND)
+            npm_run(NPM_SET_ENGINE_STRICT_COMMAND)
+          end
+
+          def ensure_environment
+            return if defined?(@environment_checked)
+            @environment_checked = true
+
+            ToolVersionChecker.check_node(minimum_version: NODE_MIN_VERSION)
+            ToolVersionChecker.check_npm(minimum_version: NPM_MIN_VERSION)
+          end
+
           private
 
+          def npm_run(cmd)
```

Comment:
Nit on naming, the command to be run doesn't need to be related to `npm`. This helper is still good because it ensures the environment is setup, but maybe `def cmd_run(cmd)` makes more sense?

Reply:
Hmmm, this command is specific to `npm` because it checks that the node and npm versions are acceptable before running them. I think we should leave it as-is.

Reply:
The point I was trying to make was that I think this could be a confusing interface. You could call `npm_run("mv foo bar")` and that would execute fine. The command isn't running an npm command, but it just calls another method which happens to have a side affect that ensures that npm and node are installed.

Thinking about it, `npm_run` is probably more clear than `cmd_run`, but probably less clear than something like `run_cmd_with_valid_node_env` but thats loud and tedious 🙈 I'm fine with whatever you want, but I wanted to call out that distinction.